gdx/src/com/badlogic/gdx/maps/objects/TextMapObject.java

```diff
@@ -0,0 +1,269 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.maps.objects;
+
+import com.badlogic.gdx.maps.MapObject;
+import com.badlogic.gdx.utils.Align;
+
+/** @brief Represents a text map object */
+public class TextMapObject extends MapObject {
```

Comment:
I am not 100% sure but I think Text objects are always rectangles? If so, I'd vote to extend `RectangleMapObject`. One more reason is to still support LibKTX `shape` extension ([link](https://github.com/libktx/ktx/blob/master/tiled/src/main/kotlin/ktx/tiled/mapObjects.kt)) without the need to change anything there.

Reply:
Can do! Though doing so, we immediately break the TiledMapObjectLoadingTest. Which is fine by me as long as it's ok for me to modify how we do the object checks.
Currently in the test they using instanceof. So the by extending from RectangleMapObject it get's picked up in there instead.

 ```
else if (mapObject instanceof ) {
				shapeRenderer.begin(ShapeRenderer.ShapeType.Filled);
				Rectangle rectangle = ((RectangleMapObject)mapObject).getRectangle();
				shapeRenderer.rect(rectangle.x, rectangle.y, rectangle.width, rectangle.height);
				shapeRenderer.end();
```

Also, I had no idea this LibKTX shape extension existed. I will take a look into it later. Also to note, there is another PR in the pipeline. Someone added a PointMapObject #6384. I'm assuming we may have to modify LibKTX shape extension as well to keep it all in sync?

Reply:
I see the point of having it separate. Then let's keep it separate to be consistent and not break an existing test. I can do the PR in LibKTX for the new object.

Reply:
Hmm, yea, looking through all the map objects even the Ellipse and Circle were made completely separated.
If we kept it as is, we could always have it maintain own Rectangle instance. Would there a need for that? Or should we just stick to having it keep track of all it's properties?

Reply:
Having its own rectangle instance with a getter would be great for the shape extension for LibKTX. I think it makes sense because an object is always related to a specific shape in Tiled like point, ellipse, rectangle or polygon.

Have a look at the RectangleMapObject and implement it in a similar way in the new text object.

Reply:
I refactored the TextMapObject a bit to be better in line with the other MapObjects. You're right in that all tiled objects do have a core "shape" to them and in the case of Text, that's obviously the object's bounds. So it makes sense it has it's own rectangle. 
Also, I removed all the Align.class related code. Convenient or not. Since it's not 1 to 1 with the Align.class. There is no point to force a relationship, especially since Tiled could easily change it at some point.
This class is now basically just a bunch of data representing the text object and it's properties. So it's up to the user to do with it as they need. 
Hope that's good. Let me know if I should alter anything else.